client/src/modules/dashboard/Events/pages/EventsPage.tsx

```diff
@@ -19,12 +21,18 @@ export const EventsPage: NextPageWithLayout = () => {
 
   const isLoading = loading || !data;
   if (isLoading || error) return <DashboardLoading error={error} />;
+  const hasPermissionToCreateEvent = checkPermission(
+    user,
+    Permission.EventCreate,
+  );
+
+  const hasPermissionToEditEvent = checkPermission(user, Permission.EventEdit);
```

Comment:
Most of these checks need to have either `eventId` or `chapterId` passed. Otherwise only check will be done only on the instance permissions.

Reply:
For current roles that would be enough. But to give ability to add custom and more granular role permissions, I think it would be better to pass the event id and/or chapter id to `checkPermission` as well. Even if so far these permissions would be only on some instance roles.

Reply:
Last question 👍, do I add the ability to filter the data that aren't editable in this PR?